### PR TITLE
Allow to set relativeTo when redirecting

### PR DESCRIPTION
**Pull request by benoitzohar**
### Description

Enable us to not be constrained by the existing router prefix when redirecting.

### CODE REVIEW

File: packages/react-router/src/hooks/redirect.ts
Comment (lemonmade) [inline packages/react-router/src/hooks/redirect.ts:6]: I'd like the second option to be an options object so it matches the `navigate` signature, what do you think of that?

File: packages/react-router/src/types.ts
Comment (lemonmade) [inline packages/react-router/src/types.ts:61]: I think we should just make this `'root' | 'prefix'`. `root` gives the new behavior, `prefix` gives the existing behavior. I think I will eventually add a third option, `'current'`, to make the navigation relative to the full current path, but for now I think just supporting the two options, with room for the third, is great.

**Review by lemonmade — CHANGES_REQUESTED** [review]
Really great job with this 👍 I have a few change requests but they are very minor

File: packages/react-router/src/hooks/redirect.ts
Comment (benoitzohar) [reply packages/react-router/src/hooks/redirect.ts:6]: Do you mean have something like that: ?

```suggestion
export function useRedirect(to: NavigateTo, {replace, relativeTo} = {replace = false}) {
```

Comment (benoitzohar) [reply packages/react-router/src/hooks/redirect.ts]: Done (I think ;))

File: packages/react-router/src/hooks/redirect.ts
Comment (lemonmade) [inline packages/react-router/src/hooks/redirect.ts:9]: We shouldn't include the `replace` option, because only `replace: true` makes sense for a redirect (otherwise, this is what I was looking for, yeah :)

**Review by lemonmade — APPROVED** [review]

**Comment by lemonmade** [comment]
Looks great 👍 Once the one little change is made, merge it in, and I will cut the new version

File: packages/react-router/src/hooks/redirect.ts
Comment (benoitzohar) [reply packages/react-router/src/hooks/redirect.ts:9]: oh yeah 🤦

**Comment by benoitzohar** [comment]
Done! I can't merge it in thought, I don't have access 😉

**Pull request merged**
